--- a/doc/udisks2-docs.xml.in.in
+++ b/doc/udisks2-docs.xml.in.in
@@ -233,6 +233,10 @@
                 <entry>If the object is a block device, this property is set to the value of the <link linkend="gdbus-property-org-freedesktop-UDisks2-Block.PreferredDevice">Block:PreferredDevice</link> property. If set, this is guaranteed to be a device file, for example <quote>/dev/vg_lucifer/lv_root</quote> or <quote>/dev/sda1</quote>. If the object is not a block device, this is not set.</entry>
               </row>
               <row>
+                <entry><parameter>device.name</parameter></entry>
+                <entry>The value of this property is set to the best display name for this device, which may be the device's <link linkend="gdbus-property-org-freedesktop-UDisks2-Block.IdLabel">Block:IdLabel</link>, or the capacity name calculated from the partition size, or the name associated with the encrypted volume. If none of the above information is present, the data assigned to the <parameter>drive</parameter> will be used as the fallback solution.</entry>
+              </row>
+              <row>
                 <entry><parameter>drive</parameter></entry>
                 <entry>Like the <parameter>device</parameter> variable, but if the object is also a drive, this variable includes Vital Product Data about the drive such as the vendor and model identifiers (if available), for example <quote>INTEL SSDSA2MH080G1GC (/dev/sda1)</quote>. Otherwise is just set to the same value as <parameter>device</parameter>. If the object is not a block device, this is not set (it is however set if the object is a block device but not a drive).</entry>
               </row>
--- a/modules/lsm/lsm_linux_drive_local.c
+++ b/modules/lsm/lsm_linux_drive_local.c
@@ -275,7 +275,7 @@
                                                      UDISKS_OBJECT (block_object),
                                                      LSM_POLICY_ACTION_ID,
                                                      options,
-                                                     N_("Authentication is required to change $(drive) LED"),
+                                                     N_("Authentication is required to change $(device.name) LED"),
                                                      invocation))
     goto out;
 
--- a/src/udisksdaemonutil.c
+++ b/src/udisksdaemonutil.c
@@ -690,6 +690,62 @@
   return ret;
 }
 
+static gchar *
+get_device_display_name (UDisksBlock *block)
+{
+  gchar *display_name = NULL;
+  gchar *s = NULL;
+  const gchar *id_label = NULL;
+  const gchar *id_usage = NULL;
+  const gchar *id_type = NULL;
+
+  if (block == NULL)
+    goto out;
+
+  id_label = udisks_block_get_id_label (block);
+  id_usage = udisks_block_get_id_usage (block);
+  id_type = udisks_block_get_id_type (block);
+  if (id_label != NULL && strlen (id_label) > 0)
+    {
+      display_name = g_strdup (id_label);
+    }
+  else if (g_strcmp0 (id_usage, "crypto") == 0)
+    {
+      s = udisks_client_get_size_for_display (NULL, udisks_block_get_size (block), FALSE, FALSE);
+      if (g_strcmp0 (id_type, "crypto_unknown") == 0)
+        {
+          /* Translators: This is used for possibly encrypted volumes.
+          *              The first %s is the formatted size (e.g. "42.0 MB").
+          */
+          display_name = g_strdup_printf (N_("%s Possibly Encrypted"), s);
+        }
+      else
+        {
+          /* Translators: This is used for encrypted volumes.
+          *              The first %s is the formatted size (e.g. "42.0 MB").
+          */
+          display_name = g_strdup_printf (N_("%s Encrypted"), s);
+        }
+      g_free (s);
+    }
+  else
+    {
+      guint64 size = udisks_block_get_size (block);
+      if (size > 0)
+        {
+          s = udisks_client_get_size_for_display (NULL, size, FALSE, FALSE);
+          /* Translators: This is used for volume with no filesystem label.
+          *              The first %s is the formatted size (e.g. "42.0 MB").
+          */
+          display_name = g_strdup_printf (N_("%s Volume"), s);
+          g_free (s);
+        }
+    }
+
+  out:
+  return display_name;
+}
+
 /**
  * udisks_daemon_util_check_authorization_sync:
  * @daemon: A #UDisksDaemon.
@@ -716,8 +772,9 @@
  * can be used in @message but note that not all variables can be used
  * in all checks. For example, any check involving a #UDisksDrive or a
  * #UDisksBlock object can safely include the fragment
- * <quote>$(drive)</quote> since it will always expand to the name of
- * the drive, e.g. <quote>INTEL SSDSA2MH080G1GC (/dev/sda1)</quote> or
+ * <quote>$(device.name)</quote> since it will always expand to the name of
+ * the partition or drive, e.g. <quote>MyUDisks (/dev/sda1)</quote> or
+ * <quote>INTEL SSDSA2MH080G1GC (/dev/sda1)</quote> or
  * the block device file e.g. <quote>/dev/vg_lucifer/lv_root</quote>
  * or <quote>/dev/sda1</quote>. However this won't work for operations
  * that isn't on a drive or block device, for example calls on the
@@ -774,6 +831,7 @@
   gboolean auth_no_user_interaction = FALSE;
   const gchar *details_device = NULL;
   gchar *details_drive = NULL;
+  gchar *device_display_name = NULL;
 
   authority = udisks_daemon_get_authority (daemon);
   if (authority == NULL)
@@ -885,6 +943,14 @@
 
   if (block != NULL)
     {
+      gchar *s = NULL;
+      s = get_device_display_name (block);
+      if (s)
+        {
+          device_display_name = g_strdup_printf ("%s (%s)", s, details_device);
+          g_free (s);
+        }
+
       _safe_polkit_details_insert (details, "id.type",    udisks_block_get_id_type (block));
       _safe_polkit_details_insert (details, "id.usage",   udisks_block_get_id_usage (block));
       _safe_polkit_details_insert (details, "id.version", udisks_block_get_id_version (block));
@@ -909,6 +975,9 @@
     polkit_details_insert (details, "device", details_device);
   if (details_drive != NULL)
     polkit_details_insert (details, "drive", details_drive);
+  if (device_display_name == NULL)
+    device_display_name = g_strdup (details_drive);
+  polkit_details_insert (details, "device.name", device_display_name);
 
   sub_error = NULL;
   result = polkit_authority_check_authorization_sync (authority,
@@ -962,6 +1031,7 @@
 
  out:
   g_free (details_drive);
+  g_free (device_display_name);
   g_clear_object (&block_object);
   g_clear_object (&drive_object);
   g_clear_object (&block);
--- a/src/udiskslinuxblock.c
+++ b/src/udiskslinuxblock.c
@@ -2894,10 +2894,10 @@
        * requests erasing a hard disk using the SECURE ERASE UNIT
        * command.
        *
-       * Do not translate $(drive), it's a placeholder and
+       * Do not translate $(device.name), it's a placeholder and
        * will be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to perform a secure erase of $(drive)");
+      message = N_("Authentication is required to perform a secure erase of $(device.name)");
       action_id = "org.freedesktop.udisks2.ata-secure-erase";
     }
   else
@@ -2906,10 +2906,10 @@
        * device. This includes both creating a filesystem or partition
        * table.
        *
-       * Do not translate $(drive), it's a placeholder and will
+       * Do not translate $(device.name), it's a placeholder and will
        * be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to format $(drive)");
+      message = N_("Authentication is required to format $(device.name)");
       action_id = format_extra_args ? "org.freedesktop.udisks2.modify-device-system" :
                                       "org.freedesktop.udisks2.modify-device";
       if (!udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
@@ -3867,10 +3867,10 @@
                                                     /* Translators: Shown in authentication dialog when creating a
                                                      * disk image file.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and will
+                                                     * Do not translate $(device.name), it's a placeholder and will
                                                      * be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to open $(drive) for reading"),
+                                                    N_("Authentication is required to open $(device.name) for reading"),
                                                     invocation))
     goto out;
 
@@ -3938,10 +3938,10 @@
                                                     /* Translators: Shown in authentication dialog when restoring
                                                      * from a disk image file.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and will
+                                                     * Do not translate $(device.name), it's a placeholder and will
                                                      * be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to open $(drive) for writing"),
+                                                    N_("Authentication is required to open $(device.name) for writing"),
                                                     invocation))
     goto out;
 
@@ -4012,10 +4012,10 @@
                                                     /* Translators: Shown in authentication dialog when an application
                                                      * wants to benchmark a device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and will
+                                                     * Do not translate $(device.name), it's a placeholder and will
                                                      * be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to open $(drive) for benchmarking"),
+                                                    N_("Authentication is required to open $(device.name) for benchmarking"),
                                                     invocation))
     goto out;
 
@@ -4095,10 +4095,10 @@
                                                     /* Translators: Shown in authentication dialog when an application
                                                      * wants to benchmark a device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and will
+                                                     * Do not translate $(device.name), it's a placeholder and will
                                                      * be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to open $(drive)."),
+                                                    N_("Authentication is required to open $(device.name)."),
                                                     invocation))
     goto out;
 
@@ -4152,10 +4152,10 @@
   /* Translators: Shown in authentication dialog when an application
    * wants to rescan a device.
    *
-   * Do not translate $(drive), it's a placeholder and will
+   * Do not translate $(device.name), it's a placeholder and will
    * be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to rescan $(drive)");
+  message = N_("Authentication is required to rescan $(device.name)");
   action_id = "org.freedesktop.udisks2.rescan";
 
   if (!udisks_daemon_util_check_authorization_sync (daemon,
--- a/src/udiskslinuxdrive.c
+++ b/src/udiskslinuxdrive.c
@@ -1075,10 +1075,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests ejecting media from a drive.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to eject $(drive)");
+  message = N_("Authentication is required to eject $(device.name)");
   action_id = "org.freedesktop.udisks2.eject-media";
   if (udisks_block_get_hint_system (block))
     {
@@ -1163,10 +1163,10 @@
   /* Translators: Shown in authentication dialog when the user
    * changes settings for a drive.
    *
-   * Do not translate $(drive), it's a placeholder and will be
+   * Do not translate $(device.name), it's a placeholder and will be
    * replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to configure settings for $(drive)");
+  message = N_("Authentication is required to configure settings for $(device.name)");
   action_id = "org.freedesktop.udisks2.modify-drive-settings";
 
   /* Check that the user is actually authorized */
@@ -1478,10 +1478,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests ejecting media from a drive.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to power off $(drive)");
+  message = N_("Authentication is required to power off $(device.name)");
   action_id = "org.freedesktop.udisks2.power-off-drive";
   if (udisks_block_get_hint_system (block))
     {
--- a/src/udiskslinuxdriveata.c
+++ b/src/udiskslinuxdriveata.c
@@ -798,10 +798,10 @@
   /* Translators: Shown in authentication dialog when the user
    * refreshes SMART data from a disk.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to update SMART data from $(drive)");
+  message = N_("Authentication is required to update SMART data from $(device.name)");
   action_id = "org.freedesktop.udisks2.ata-smart-update";
 
   if (atasmart_blob != NULL)
@@ -809,10 +809,10 @@
       /* Translators: Shown in authentication dialog when the user
        * tries to simulate SMART data from a libatasmart blob.
        *
-       * Do not translate $(drive), it's a placeholder and
+       * Do not translate $(device.name), it's a placeholder and
        * will be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to set SMART data from a blob on $(drive)");
+      message = N_("Authentication is required to set SMART data from a blob on $(device.name)");
       action_id = "org.freedesktop.udisks2.ata-smart-simulate";
     }
   else
@@ -943,10 +943,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * aborts a running SMART self-test.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to abort a SMART self-test on $(drive)"),
+                                                    N_("Authentication is required to abort a SMART self-test on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -1191,10 +1191,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * initiates a SMART self-test.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to start a SMART self-test on $(drive)"),
+                                                    N_("Authentication is required to start a SMART self-test on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -1331,10 +1331,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests the power state of a drive.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to check power state for $(drive)");
+  message = N_("Authentication is required to check power state for $(device.name)");
   action_id = "org.freedesktop.udisks2.ata-check-power";
 
   /* TODO: maybe not check with polkit if this is OK (consider gnome-disks(1) polling all drives every few seconds) */
@@ -1427,11 +1427,11 @@
    * tries to wake up a drive from standby mode or tries to put a drive into
    * standby mode.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = (do_wakeup) ? N_("Authentication is required to wake up $(drive) from standby mode") :
-                          N_("Authentication is required to put $(drive) in standby mode");
+  message = (do_wakeup) ? N_("Authentication is required to wake up $(device.name) from standby mode") :
+                          N_("Authentication is required to put $(device.name) in standby mode");
   action_id = "org.freedesktop.udisks2.ata-standby";
   if (udisks_block_get_hint_system (block))
     {
@@ -2287,10 +2287,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests erasing a hard disk using the SECURE ERASE UNIT command.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to perform a secure erase of $(drive)");
+  message = N_("Authentication is required to perform a secure erase of $(device.name)");
   action_id = "org.freedesktop.udisks2.ata-secure-erase";
 
   /* Check that the user is authorized */
@@ -2379,20 +2379,20 @@
       /* Translators: Shown in authentication dialog when the user
        * requests enabling SMART on a disk.
        *
-       * Do not translate $(drive), it's a placeholder and
+       * Do not translate $(device.name), it's a placeholder and
        * will be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to enable SMART on $(drive)");
+      message = N_("Authentication is required to enable SMART on $(device.name)");
     }
   else
     {
       /* Translators: Shown in authentication dialog when the user
        * requests enabling SMART on a disk.
        *
-       * Do not translate $(drive), it's a placeholder and
+       * Do not translate $(device.name), it's a placeholder and
        * will be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to disable SMART on $(drive)");
+      message = N_("Authentication is required to disable SMART on $(device.name)");
     }
   action_id = "org.freedesktop.udisks2.ata-smart-enable-disable";
 
--- a/src/udiskslinuxencrypted.c
+++ b/src/udiskslinuxencrypted.c
@@ -492,10 +492,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests unlocking an encrypted device.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to unlock the encrypted device $(drive)");
+  message = N_("Authentication is required to unlock the encrypted device $(device.name)");
   if (!udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (is_in_crypttab && has_option (crypttab_options, "x-udisks-auth"))
@@ -772,10 +772,10 @@
                                                                     * requests locking an encrypted device that was previously.
                                                                     * unlocked by another user.
                                                                     *
-                                                                    * Do not translate $(drive), it's a placeholder and
+                                                                    * Do not translate $(device.name), it's a placeholder and
                                                                     * will be replaced by the name of the drive/device in question
                                                                     */
-                                                                   N_("Authentication is required to lock the encrypted device $(drive) unlocked by another user"),
+                                                                   N_("Authentication is required to lock the encrypted device $(device.name) unlocked by another user"),
                                                                    invocation,
                                                                    error))
         {
@@ -960,10 +960,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * requests unlocking an encrypted device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to unlock the encrypted device $(drive)"),
+                                                    N_("Authentication is required to unlock the encrypted device $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -1094,10 +1094,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests resizing a encrypted block device.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to resize the encrypted device $(drive)");
+  message = N_("Authentication is required to resize the encrypted device $(device.name)");
   if (! udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
--- a/src/udiskslinuxfilesystem.c
+++ b/src/udiskslinuxfilesystem.c
@@ -932,10 +932,10 @@
       /* Translators: Shown in authentication dialog when the user
        * requests mounting a filesystem.
        *
-       * Do not translate $(drive), it's a placeholder and
+       * Do not translate $(device.name), it's a placeholder and
        * will be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to mount $(drive)");
+      message = N_("Authentication is required to mount $(device.name)");
       if (mount_other_user)
         {
           action_id = "org.freedesktop.udisks2.filesystem-mount-other-user";
@@ -1018,13 +1018,13 @@
                * user requests mounting a filesystem that is in
                * /etc/fstab file with the x-udisks-auth option.
                *
-               * Do not translate $(drive), it's a
+               * Do not translate $(device.name), it's a
                * placeholder and will be replaced by the name of
                * the drive/device in question
                *
                * Do not translate /etc/fstab
                */
-              message = N_("Authentication is required to mount $(drive) referenced in the /etc/fstab file");
+              message = N_("Authentication is required to mount $(device.name) referenced in the /etc/fstab file");
               if (!udisks_daemon_util_check_authorization_sync (daemon,
                                                                 object,
                                                                 action_id,
@@ -1123,10 +1123,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests mounting a filesystem.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to mount $(drive)");
+  message = N_("Authentication is required to mount $(device.name)");
   if (mount_other_user)
     {
       action_id = "org.freedesktop.udisks2.filesystem-mount-other-user";
@@ -1599,13 +1599,13 @@
                                                                  * user requests unmounting a filesystem that is in
                                                                  * /etc/fstab file with the x-udisks-auth option.
                                                                  *
-                                                                 * Do not translate $(drive), it's a
+                                                                 * Do not translate $(device.name), it's a
                                                                  * placeholder and will be replaced by the name of
                                                                  * the drive/device in question
                                                                  *
                                                                  * Do not translate /etc/fstab
                                                                  */
-                                                                N_("Authentication is required to unmount $(drive) referenced in the /etc/fstab file"),
+                                                                N_("Authentication is required to unmount $(device.name) referenced in the /etc/fstab file"),
                                                                 invocation))
                 goto out;
               unmount_fstab_as_root = TRUE;
@@ -1650,10 +1650,10 @@
        * requests unmounting a filesystem previously mounted by
        * another user.
        *
-       * Do not translate $(drive), it's a placeholder and
+       * Do not translate $(device.name), it's a placeholder and
        * will be replaced by the name of the drive/device in question
        */
-      message = N_("Authentication is required to unmount $(drive) mounted by another user");
+      message = N_("Authentication is required to unmount $(device.name) mounted by another user");
 
       if (!udisks_daemon_util_check_authorization_sync (daemon,
                                                         object,
@@ -1818,10 +1818,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests changing the filesystem label.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to change the filesystem label on $(drive)");
+  message = N_("Authentication is required to change the filesystem label on $(device.name)");
   if (!udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
@@ -2000,10 +2000,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests resizing the filesystem.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to resize the filesystem on $(drive)");
+  message = N_("Authentication is required to resize the filesystem on $(device.name)");
   if (! udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
@@ -2171,10 +2171,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests resizing the filesystem.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to repair the filesystem on $(drive)");
+  message = N_("Authentication is required to repair the filesystem on $(device.name)");
   if (! udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
@@ -2339,10 +2339,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests resizing the filesystem.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to check the filesystem on $(drive)");
+  message = N_("Authentication is required to check the filesystem on $(device.name)");
   if (! udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
@@ -2499,10 +2499,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests taking ownership of the filesystem.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to change ownership of the filesystem on $(drive)");
+  message = N_("Authentication is required to change ownership of the filesystem on $(device.name)");
 
   /* Check that the user is actually authorized to check the filesystem. */
   if (! udisks_daemon_util_check_authorization_sync (daemon,
@@ -2666,10 +2666,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests changing the filesystem UUID.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to change the filesystem UUID on $(drive)");
+  message = N_("Authentication is required to change the filesystem UUID on $(device.name)");
   if (!udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
--- a/src/udiskslinuxloop.c
+++ b/src/udiskslinuxloop.c
@@ -237,10 +237,10 @@
                                                          * requests deleting a loop device previously set up by
                                                          * another user.
                                                          *
-                                                         * Do not translate $(drive), it's a placeholder and
+                                                         * Do not translate $(device.name), it's a placeholder and
                                                          * will be replaced by the name of the drive/device in question
                                                          */
-                                                        N_("Authentication is required to delete the loop device $(drive)"),
+                                                        N_("Authentication is required to delete the loop device $(device.name)"),
                                                         invocation))
         goto out;
     }
@@ -330,10 +330,10 @@
                                                          * requests changing autoclear on a loop device set up by
                                                          * another user.
                                                          *
-                                                         * Do not translate $(drive), it's a placeholder and
+                                                         * Do not translate $(device.name), it's a placeholder and
                                                          * will be replaced by the name of the drive/device in question
                                                          */
-                                                        N_("Authentication is required to modify the loop device $(drive)"),
+                                                        N_("Authentication is required to modify the loop device $(device.name)"),
                                                         invocation))
         goto out;
     }
--- a/src/udiskslinuxnvmecontroller.c
+++ b/src/udiskslinuxnvmecontroller.c
@@ -460,10 +460,10 @@
   /* Translators: Shown in authentication dialog when the user
    * refreshes SMART data from a disk.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to update SMART data from $(drive)");
+  message = N_("Authentication is required to update SMART data from $(device.name)");
   action_id = "org.freedesktop.udisks2.nvme-smart-update";
 
   /* Check that the user is authorized */
@@ -822,10 +822,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * initiates a device self-test.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to start a device self-test on $(drive)"),
+                                                    N_("Authentication is required to start a device self-test on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -915,10 +915,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * aborts a running device self-test.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to abort a device self-test on $(drive)"),
+                                                    N_("Authentication is required to abort a device self-test on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -1217,10 +1217,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * initiates a sanitize operation.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to perform a sanitize operation of $(drive)"),
+                                                    N_("Authentication is required to perform a sanitize operation of $(device.name)"),
                                                     invocation))
     goto out;
 
--- a/src/udiskslinuxnvmefabrics.c
+++ b/src/udiskslinuxnvmefabrics.c
@@ -195,10 +195,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests disconnect of a NVMeoF connected controller.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to disconnect a NVMe over Fabrics controller $(drive)");
+  message = N_("Authentication is required to disconnect a NVMe over Fabrics controller $(device.name)");
   action_id = "org.freedesktop.udisks2.nvme-disconnect";
 
   /* Check that the user is authorized */
--- a/src/udiskslinuxnvmenamespace.c
+++ b/src/udiskslinuxnvmenamespace.c
@@ -403,10 +403,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * initiates a device self-test.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to format a namespace on $(drive)"),
+                                                    N_("Authentication is required to format a namespace on $(device.name)"),
                                                     invocation))
     goto out;
 
--- a/src/udiskslinuxpartition.c
+++ b/src/udiskslinuxpartition.c
@@ -143,10 +143,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests modifying a partition (changing type, flags, name etc.).
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to modify the partition on device $(drive)");
+  message = N_("Authentication is required to modify the partition on device $(device.name)");
   if (!udisks_daemon_util_setup_by_user (daemon, object, *caller_uid))
     {
       if (udisks_block_get_hint_system (block))
--- a/src/udiskslinuxpartitiontable.c
+++ b/src/udiskslinuxpartitiontable.c
@@ -333,10 +333,10 @@
   /* Translators: Shown in authentication dialog when the user
    * requests creating a new partition.
    *
-   * Do not translate $(drive), it's a placeholder and
+   * Do not translate $(device.name), it's a placeholder and
    * will be replaced by the name of the drive/device in question
    */
-  message = N_("Authentication is required to create a partition on $(drive)");
+  message = N_("Authentication is required to create a partition on $(device.name)");
   if (!udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
     {
       if (udisks_block_get_hint_system (block))
--- a/src/udiskslinuxswapspace.c
+++ b/src/udiskslinuxswapspace.c
@@ -190,10 +190,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * requests activating a swap device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to activate swapspace on $(drive)"),
+                                                    N_("Authentication is required to activate swapspace on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -288,10 +288,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * requests deactivating a swap device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to deactivate swapspace on $(drive)"),
+                                                    N_("Authentication is required to deactivate swapspace on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -365,10 +365,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * requests setting label of a swap device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to set swapspace label on $(drive)"),
+                                                    N_("Authentication is required to set swapspace label on $(device.name)"),
                                                     invocation))
     goto out;
 
@@ -451,10 +451,10 @@
                                                     /* Translators: Shown in authentication dialog when the user
                                                      * requests setting UUID of a swap device.
                                                      *
-                                                     * Do not translate $(drive), it's a placeholder and
+                                                     * Do not translate $(device.name), it's a placeholder and
                                                      * will be replaced by the name of the drive/device in question
                                                      */
-                                                    N_("Authentication is required to set swapspace UUID on $(drive)"),
+                                                    N_("Authentication is required to set swapspace UUID on $(device.name)"),
                                                     invocation))
     goto out;
 
